Add buttons to checker dialogs

windymilla opened the pull request:
There are either 2 or 6 buttons, depending on whether the checker is able to Fix a reported problem. The buttons should resize with the dialog width.

Each button has a tooltip, and clicking on each one is equivalent to using Ctrl/Cmd/Shift Left/Right mouse buttons in the message list, but just using a simple mouse click.

Fixes #752

I have decided for now not to add a Preference setting to control whether these buttons are shown.

srjfoo commented:
Looking at
<img width="828" alt="image" src="https://github.com/user-attachments/assets/ff123926-4e94-4e70-a2d9-3a760cc10390" />
(the spell checker) and comparing the tooltip with that last row of buttons, I don't think that last row should even be there. "Fix" doesn't make sense in the spell checker, does it? All of the buttons in the second row correspond to the actions in the tooltip, don't they? Or am I missing something?

windymilla commented:
> comparing the tooltip with that last row of buttons, I don't think that last row should even be there. "Fix" doesn't make sense in the spell checker, does it? All of the buttons in the second row correspond to the actions in the tooltip, don't they? Or am I missing something?

Yes and no ;)

The whole point of the last row of buttons is precisely to provide equivalent buttons that correspond to the actions in the tooltip. In general for checker dialogs this is to support users for whom holding Ctrl/Cmd/Shift while clicking the mouse is not easy.

However, in this specific dialog, the key point in favor of your suggestion to remove the last row of buttons is that they are
(poorly worded) equivalents of the buttons on the row above:
Remove = Skip
Remove All = Skip All
Variants of Fix = Add to Proj Dict

So, I've disabled the "generic" buttons in the Spell Dialog. I think they are probably relevant in the other checker dialogs though.

srjfoo commented:
Part 1 of nits (to be continued, because it's time to stop for the night, but I should go ahead and ask about these):
* I'd rather see "Remove" spelled out in "Rem. All". Most of the buttons have everything spelled out, even with text that is much longer than "Remove All" would be.
* If you have to abbreviate button labels (and "Prev." and "Col" are good places to), standardize: either punctuate all abbreviations, or don't punctuate any.
* Actually, "Hide" makes more sense to me than "Remove" but I suppose "Remove" has the weight of "Tradition" behind it, so ... 🤷‍♀️ 😁 
* Curly Quotes check doesn't have the "shift" version of the tooltip when hovering over the results, but has the buttons for it.
* I know that I didn't like the generic buttons in the spelling checker, especially because there were already custom buttons there, but I just noticed that the custom buttons say "skip" and "skip all" instead of "Remove" and "Remove All" (the tooltip uses "Remove"). Should they be one or the other?

windymilla commented:
> Part 1 of nits (to be continued, because it's time to stop for the night, but I should go ahead and ask about these):
> 
> * I'd rather see "Remove" spelled out in "Rem. All". Most of the buttons have everything spelled out, even with text that is much longer than "Remove All" would be.
> * If you have to abbreviate button labels (and "Prev." and "Col" are good places to), standardize: either punctuate all abbreviations, or don't punctuate any.
> * Actually, "Hide" makes more sense to me than "Remove" but I suppose "Remove" has the weight of "Tradition" behind it, so ... 🤷‍♀️ 😁
> * Curly Quotes check doesn't have the "shift" version of the tooltip when hovering over the results, but has the buttons for it.
> * I know that I didn't like the generic buttons in the spelling checker, especially because there were already custom buttons there, but I just noticed that the custom buttons say "skip" and "skip all" instead of "Remove" and "Remove All" (the tooltip uses "Remove"). Should they be one or the other?

* I like the word Hide - it's a better description of what is happening, and is shorter
* Removed period after "Prev." & "Chap."
* Changed Remove to hide in many tooltips
* Changed Remove to Skip for the Spelling dialog
